### PR TITLE
feat(location): filter location by account project [IR-3890]

### PR DESCRIPTION
**Pull request by mikeplascdev**
## Summary

- Get location by account

## References
closes #_insert number here_

## Jira
[IR-3890](https://tsu.atlassian.net/browse/IR-3890)

## Evidence

https://github.com/user-attachments/assets/77bb7c34-a012-44b5-8892-1a13efb97dd9

### CODE REVIEW

**Comment by jonathan-casarrubias** [comment]
This should be implemented in: packages/projects/projects/theinfinitereality/irpro-multitenancy/services/hooks/social/location.hooks.ts

It's a MT specific feature

File: packages/server-core/src/social/location/location.hooks.ts [hunk 1]
```diff
@@ -43,6 +43,9 @@ import { UserID } from '@ir-engine/common/src/schemas/user/user.schema'
 import verifyScope from '@ir-engine/server-core/src/hooks/verify-scope'
 
 import { projectHistoryPath, staticResourcePath } from '@ir-engine/common/src/schema.type.module'
+import isAction from '@ir-engine/server-core/src/hooks/is-action'
+import makeQueryJoinable from '@ir-engine/server-core/src/hooks/make-query-joinable'
+import { accountProjectPath } from '@theinfinitereality/irpro-multitenancy/services/account/account-project/account-project.schema'
```
Comment (HexaField) [inline packages/server-core/src/social/location/location.hooks.ts:48]: we cannot import any multitenancy code in the core engine

File: packages/server-core/src/social/location/location.hooks.ts [hunk 2]
```diff
@@ -86,6 +89,23 @@ const sortByLocationSetting = async (context: HookContext<LocationService>) => {
   }
 }
 
+/**
+ * Hook used to ensure all account projects that a user has permission to are returned for location.
+ * @param context
+ * @returns
+ */
+export const locationByAccountProject = async (context: HookContext) => {
```
Comment (HexaField) [inline packages/server-core/src/social/location/location.hooks.ts:97]: this hook should injected in multitenancy

**Review by HexaField — CHANGES_REQUESTED** [review]

**Comment by HexaField** [comment]
Please make these changes in MT

**Pull request closed without merging**